### ISSUE TITLE
Error during the Get junit SLA report

### ISSUE THREAD

**Issue by bcouavoux** [Neotys-Labs/neoload-cli#116]
Hello,

We have an error when we try to retrieve the sla report via neoload-cli in our pipeline :

**The pipeline :**

- task: Bash@3
    displayName: Install NeoLoad
    continueOnError: false
    inputs:
      targetType: 'inline'
      script: |
        pip install neoload
        neoload --version

- task: Bash@3
    displayName: Configure the NeoLoad test
    continueOnError: true
    inputs:
      targetType: 'inline'
      script: |
         neoload login --url "${{variables.neoloadApiUrl}}" --workspace "${{variables.workspaceName}}" "${{parameters.neoloadToken}}" \
         test-settings --zone "${{parameters.zoneId}}" --lgs ${{parameters.lgs}} --scenario "${{parameters.scenario}}" createorpatch "${{parameters.TestNameNeoload}}" \
         project --path "$(System.DefaultWorkingDirectory)" upload "${{parameters.TestNameNeoload}}"

- task: Bash@3
    displayName: Run the NeoLoad test
    continueOnError: true
    inputs:
      targetType: 'inline'
      script: |
        neoload run \
        --as-code default.yaml \
        --name "${{parameters.TitleTest}}" --description "${{parameters.DescriptionTest}}"

- task: Bash@3
    displayName: Get the junit SLA report
    continueOnError: true
    inputs:
      targetType: 'inline'
      script: |
       neoload test-results --junit-file $(Common.TestResultsDirectory)/neoload-slas.xml junitsla
       cat $(Common.TestResultsDirectory)/neoload-slas.xml

- task: PublishTestResults@2
    displayName: Publish SLA test results to Azure Pipelines
    inputs:
      testResultsFormat: 'JUnit'
      testResultsFiles: $(Common.TestResultsDirectory)/*.xml
      failTaskOnFailedTests: true

**The result of the test with the sla summary :**

Status: INIT
Status: RUNNING
    00:00:16/00:02:00	 Err[--], LGs[1]	 VUs:8	 BPS[80340.516]	 RPS:252.000	 avg(rql): 30
    00:00:37/00:02:00	 Err[--], LGs[1]	 VUs:15	 BPS[153709.16]	 RPS:257.000	 avg(rql): 45
    00:00:57/00:02:00	 Err[--], LGs[1]	 VUs:11	 BPS[158040.44]	 RPS:154.000	 avg(rql): 51
    00:01:18/00:02:00	 Err[--], LGs[1]	 VUs:--	 BPS[158769.52]	 RPS:--	 avg(rql): 48
    00:01:39/00:02:00	 Err[--], LGs[1]	 VUs:--	 BPS[152249.17]	 RPS:--	 avg(rql): 48
    00:01:59/00:02:00	 Err[--], LGs[1]	 VUs:--	 BPS[145955.36]	 RPS:--	 avg(rql): 48
Status: TERMINATED
SLA summary:

PerIntervalSLA [avg-resp-time] WARNING on [POCNEOLOAD > Total Tickets > Ticket 1 [19.556% >= [2.0]]]
PerIntervalSLA [avg-resp-time] WARNING on [POCNEOLOAD > Total Tickets > Ticket 2 [4.889% >= [2.0]]]

**And the error during the Get the junit SLA report task :**

Traceback (most recent call last):
  File "c:\hostedtoolcache\windows\python\3.8.6\x64\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "c:\hostedtoolcache\windows\python\3.8.6\x64\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\hostedtoolcache\windows\Python\3.8.6\x64\Scripts\neoload.exe\__main__.py", line 7, in <module>
    
  File "c:\hostedtoolcache\windows\python\3.8.6\x64\lib\site-packages\click\core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "c:\hostedtoolcache\windows\python\3.8.6\x64\lib\site-packages\click\core.py", line 782, in main
    rv = self.invoke(ctx)
  File "c:\hostedtoolcache\windows\python\3.8.6\x64\lib\site-packages\click\core.py", line 1289, in invoke
    rv.append(sub_ctx.command.invoke(sub_ctx))
  File "c:\hostedtoolcache\windows\python\3.8.6\x64\lib\site-packages\click\core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "c:\hostedtoolcache\windows\python\3.8.6\x64\lib\site-packages\click\core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "c:\hostedtoolcache\windows\python\3.8.6\x64\lib\site-packages\neoload\commands\test_results.py", line 72, in cli
    junit(__id, junit_file)
  File "c:\hostedtoolcache\windows\python\3.8.6\x64\lib\site-packages\neoload\commands\test_results.py", line 117, in junit
    displayer.print_result_junit(json_result, json_sla_test, json_sla_interval, json_sla_global, junit_file)
  File "C:\hostedtoolcache\windows\Python\3.8.6\x64\Lib\site-packages\neoload\neoload_cli_lib\displayer.py", line 69, in print_result_junit
    with open(junit_file_path, 'w') as stream:

**Comment by paulsbruce** [comment]
@bcouavoux seems to be that the CLI can't open the file you specified $(Common.TestResultsDirectory)/neoload-slas.xml

But it seems like the upload based on System.DefaultWorkingDirectory has at least proper read permissions for the CLI. Can you try writing your jUnit file to the System.DefaultWorkingDirectory path? It may be some Azure-specific directory permissions thing on the test results directory. If that works, try doing a copy/cp of that file to the TestResultsDirectory as a separate step and let us know what the script and output is.

**Comment by bcouavoux** [comment]
Thanks for your reactivity !
I just tried with the the upload on the System.DefaultWorkingDirectory but the same error happened :( 
**The task :**

  - task: Bash@3
    displayName: Get the junit SLA report
    continueOnError: true
    inputs:
      targetType: 'inline'
      script: |
       neoload test-results --junit-file $(System.DefaultWorkingDirectory)/neoload-slas.xml junitsla
       cat $(System.DefaultWorkingDirectory)/neoload-slas.xml

**The error :**

Traceback (most recent call last):
  File "c:\hostedtoolcache\windows\python\3.8.6\x64\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "c:\hostedtoolcache\windows\python\3.8.6\x64\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\hostedtoolcache\windows\Python\3.8.6\x64\Scripts\neoload.exe\__main__.py", line 7, in <module>
    
  File "c:\hostedtoolcache\windows\python\3.8.6\x64\lib\site-packages\click\core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "c:\hostedtoolcache\windows\python\3.8.6\x64\lib\site-packages\click\core.py", line 782, in main
    rv = self.invoke(ctx)
  File "c:\hostedtoolcache\windows\python\3.8.6\x64\lib\site-packages\click\core.py", line 1289, in invoke
    rv.append(sub_ctx.command.invoke(sub_ctx))
  File "c:\hostedtoolcache\windows\python\3.8.6\x64\lib\site-packages\click\core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "c:\hostedtoolcache\windows\python\3.8.6\x64\lib\site-packages\click\core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "c:\hostedtoolcache\windows\python\3.8.6\x64\lib\site-packages\neoload\commands\test_results.py", line 72, in cli
    junit(__id, junit_file)
  File "c:\hostedtoolcache\windows\python\3.8.6\x64\lib\site-packages\neoload\commands\test_results.py", line 117, in junit
    displayer.print_result_junit(json_result, json_sla_test, json_sla_interval, json_sla_global, junit_file)
  File "C:\hostedtoolcache\windows\Python\3.8.6\x64\Lib\site-packages\neoload\neoload_cli_lib\displayer.py", line 69, in print_result_junit
    with open(junit_file_path, 'w') as stream:

**Comment by bcouavoux** [comment]
The pipeline is launched on an microsoft hosted agent (windows)

**Comment by paulsbruce** [comment]
@bcouavoux i just ran a test using a Ubuntu vmImage and it was fine: https://github.com/paulsbruce/neoload-ci-utility/blob/master/pipelines/azure/neoload-cli-eventshub.yml

I also notice that right now, US Hosted Pool (seems to be mostly windows agents) is having problems: https://status.dev.azure.com/

**Comment by bcouavoux** [comment]
Our Hosted pool is in on EuropeWest and dont seem to has issue .. 
Do you have any idea where this error is coming from? I can't debug the problem at all..

Can you test with an Microsoft vmimage ?

**Comment by bcouavoux** [comment]
@paulsbruce Indeed, that work with an ubuntu vmImage but we have to used an Windows image because we have a lot of task in our pipeline that requiert an Windows OS.

Can you see why it doesn't work on a windows OS pls ?

**Comment by bcouavoux** [comment]
Any news @paulsbruce ? we really need to publish the result of our sla in our pipelines and we have to use windows agents ..
If there is anything i can do to help you feel free to ask me

**Comment by paulsbruce** [comment]
@bcouavoux since I wasn't able to repro your situation with my account and scripts, I think a call is best. DM me over [linkedin](https://www.linkedin.com/in/paulsbruce/) or [twitter](https://twitter.com/paulsbruce) with your email, then we can set up a session to diagnose what's going on. Afterward, we'll update this issue with the generic resolution (minus private details) for transparency's sake.

**Comment by bcouavoux** [comment]
Yes thank you !
i cant dm you over linkedin because i dont have linkedin premium and your dms on twitter are not allowed.
You can dm me over my [linkedin ](https://www.linkedin.com/in/baptiste-couavoux-98901914b/) as u want

**Comment by paulsbruce** [comment]
@bcouavoux What we found by looking at this together was that the path being fed to NeoLoad CLI was invalid, even if it was coming from an intrinsic environment variable. To figure this out, we:

1. put some 'pwd' and 'cd $(Agent.TempDirectory)' commands diagnostically into a copy of your pipeline script to confirm what the values actually are at runtime
2. to speed re-runs, we commented out the actual test run blocks and instead simply used an existing test result ('neoload test-settings use [ID]')
2. made sure that all tokens and variables were updated with values that work

Examples of pipeline outputs...note the differences in $(Agent.TempDirectory) and $(System.DefaultWorkingDirectory):

Generating script. 
"C:\Program Files\Git\bin\bash.exe" --noprofile --norc -c pwd 
/d/a/_temp 

"C:\Program Files\Git\bin\bash.exe" --noprofile --norc /d/a/_temp/68e9fc01-de96-487d-948b-ef14ed0076f9.sh 
mv: cannot stat 'D:a1s/neoload-slas.xml': No such file or directory
